### PR TITLE
Spotify.js tests

### PR DESCRIPTION
**Pull request by medegor44**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "axios": "^0.21.1",
```
Comment (confirmit-horizons) [inline package.json:15]: Isn't it already there? Probably need to rebase on master

File: src/utils/__tests__/spotify.spec.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    axios.mockRejectedValue(error500);
+
+    try {
+      await fetchUser("token");
+    } catch (e) {
+      expect(e).toEqual(
+        new RetriesCountExceededError("Count of retries is exceeded")
+      );
+    }
+  }, 15000);
```
Comment (confirmit-horizons) [inline src/utils/__tests__/spotify.spec.js:81]: it's not clear why timeouts needed. it's bad when unit test needs to wait a lot. you can make delays configurable as function arguments (ie `requestToApi(url, token, retryCount=10, retryDelay=500)`), then in tests you can set 1ms delay and assert if wait for 3ms when you have 3 retries etc

File: src/utils/__tests__/spotify.spec.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+});
+
+describe("tests for requestToApi", () => {
```
Comment (confirmit-horizons) [inline src/utils/__tests__/spotify.spec.js:84]: probably can be splitted to a separate test files (1 describe per spec) for better readability

**Review by confirmit-horizons — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (medegor44) [reply package.json:15]: I will create new branch from actual master and then cherry-pick commits with tests from this branch.

upd: check new PR https://github.com/medegor44/spotify-playlist/pull/22

**Pull request closed without merging**